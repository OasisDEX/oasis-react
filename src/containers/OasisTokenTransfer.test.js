/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { fromJS } from 'immutable';

import {
  OasisTokenTransferWrapper,
  mapStateToProps,
  mapDispatchToProps
} from './OasisTokenTransfer';
import { shallow } from 'enzyme';

describe('(Container) OasisTokenTransfer', () => {
  const state = fromJS(global.storeMock);
  const initialProps = mapStateToProps(state);
  const initialActions = mapDispatchToProps(x => x);
  const props = {
    ...initialActions,
    ...initialProps
  };

  it('will receive right props', () => {
    expect(initialProps).toMatchSnapshot();
  });


  it('will receive right actions', () => {
    expect(initialActions).toMatchSnapshot();
  });

  it('should render', () => {
    const wrapper = shallow(
      <OasisTokenTransferWrapper {...props}/>
    );
    expect(wrapper).toMatchSnapshot();
  });

});
