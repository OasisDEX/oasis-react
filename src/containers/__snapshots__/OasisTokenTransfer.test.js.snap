// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) OasisTokenTransfer should render 1`] = `
<OasisWidgetFrame
  heading="Transfer"
>
  <div>
    <div
      style={
        Object {
          "display": "flex",
          "marginBottom": "20px",
        }
      }
    >
      <span
        style={
          Object {
            "textTransform": "uppercase",
            "width": "50%",
          }
        }
      >
        Wallet
      </span>
      <Connect(OasisTokenBalance)
        decimalPlaces={5}
        style={
          Object {
            "width": "50%",
          }
        }
        tokenName="MKR"
      />
    </div>
    <hr />
    <div>
      <Connect(OasisTokenSelect)
        name="tokenTransfer"
      />
      <Connect(ReduxForm)
        onSubmit={[Function]}
      />
    </div>
    <div>
      <hr />
      <Connect(TokenTransferStatus) />
    </div>
  </div>
</OasisWidgetFrame>
`;

exports[`(Container) OasisTokenTransfer will receive right actions 1`] = `
Object {
  "actions": Object {
    "makeTransfer": [Function],
  },
}
`;

exports[`(Container) OasisTokenTransfer will receive right props 1`] = `
Object {
  "selectedToken": "MKR",
  "transaction": undefined,
}
`;
